ibis/pandas/execution/window.py

```diff
@@ -73,21 +73,31 @@ def _post_process_group_by_order_by(series, parent, order_by, group_by):
 
 
 @execute_node.register(ops.WindowOp, pd.Series, win.Window)
-def execute_window_op(op, data, window, scope=None, aggcontext=None, **kwargs):
+def execute_window_op(
+    op, data, window, scope=None, aggcontext=None, clients=None, **kwargs
+):
     operand = op.expr
+    # pre execute "manually" here because otherwise we wouldn't pickup
+    # relevant scope changes from the child operand since we're managing
+    # execution of that by hand
+    operand_op = operand.op()
+    pre_executed_scope = pre_execute(
+        operand_op, *clients, scope=scope, aggcontext=aggcontext, **kwargs)
+    scope = toolz.merge(scope, pre_executed_scope)
+
     root, = op.root_tables()
     root_expr = root.to_expr()
     data = execute(root_expr, scope=scope, aggcontext=aggcontext, **kwargs)
 
     following = window.following
     order_by = window._order_by
 
-    if order_by and following != 0:
+    if order_by and following != 0 and not isinstance(
+        operand_op, ops.ShiftBase
```

Comment:
For the record, this is here because `lag`/`lead` operations are not affected by `preceding` or `following`.